backend/src/services/collectionService.ts

```diff
@@ -398,142 +396,133 @@ export class CollectionService extends LoggerBase {
-          const label = parsedUrl.pathname.slice(1) // removes leading '/'
-          addToResult(PlatformType.GITHUB, url, label)
+        if (i.platform === PlatformType.GERRIT) {
+          for (const r of (i.settings as any).remote.repos) {
+            addToResult(i.platform, r, r)
+          }
         }
-      } catch (err) {
-        // Do nothing
       }
-    }
 
-    return result
+      return result
+    })
   }
 
   async findGithubInsightsForSegment(segmentId: string): Promise<IGithubInsights> {
-    const qx = SequelizeRepository.getQueryExecutor(this.options)
-    const integrations = await fetchIntegrationsForSegment(qx, segmentId)
-    const segment = await findSegmentById(qx, segmentId)
+    return SequelizeRepository.withTx(this.options, async (tx) => {
+      const qx = SequelizeRepository.getQueryExecutor(this.options, tx)
+      const integrations = await fetchIntegrationsForSegment(qx, segmentId)
+      const segment = await findSegmentById(qx, segmentId)
 
-    const [githubIntegration] = integrations.filter(
-      (integration) =>
-        integration.platform === PlatformType.GITHUB ||
-        integration.platform === PlatformType.GITHUB_NANGO,
-    )
+      const [githubIntegration] = integrations.filter(
+        (integration) =>
+          integration.platform === PlatformType.GITHUB ||
+          integration.platform === PlatformType.GITHUB_NANGO,
+      )
 
-    if (!githubIntegration) {
-      return null
-    }
+      if (!githubIntegration) {
+        return null
+      }
 
-    const settings = githubIntegration.settings as GithubIntegrationSettings
+      const settings = githubIntegration.settings as GithubIntegrationSettings
 
-    const mainOrg = await GithubIntegrationService.findMainGithubOrganizationWithLLM(
-      qx,
-      segment.name,
-      settings.orgs,
-    )
+      const mainOrg = await GithubIntegrationService.findMainGithubOrganizationWithLLM(
+        qx,
+        segment.name,
+        settings.orgs,
+      )
 
-    if (!mainOrg) {
-      return null
-    }
+      if (!mainOrg) {
+        return null
+      }
 
-    const { description, name, repos } = mainOrg
+      const { description, name, repos } = mainOrg
 
-    const orgDetail = await GithubIntegrationService.findOrgDetail(name)
+      const orgDetail = await GithubIntegrationService.findOrgDetail(name)
 
-    if (!orgDetail) {
-      return null
-    }
+      if (!orgDetail) {
+        return null
+      }
 
-    const { logoUrl, github, website, twitter } = orgDetail
+      const { logoUrl, github, website, twitter } = orgDetail
 
-    const topics = await GithubIntegrationService.findOrgTopics(name, repos)
+      const topics = await GithubIntegrationService.findOrgTopics(name, repos)
 
-    return {
-      description,
-      github,
-      logoUrl,
-      name,
-      topics,
-      twitter,
-      website,
-    }
+      return {
+        description,
+        github,
+        logoUrl,
+        name,
```

Comment:
Yesterday while I was using this feature in production. By creating projects manually, I found that using the `name` as the one that is returned on the github organization might not be ideal.
I think we should stick to the segment name.

Could we update this return payload so that:
```
name: segment.name
```
?